### PR TITLE
Problem Indexing

### PR DESCRIPTION
**Pull request by xtalax**
see https://github.com/SciML/ModelingToolkit.jl/issues/2065#issuecomment-1417899734

### CODE REVIEW

File: src/remake.jl [hunk 1]
```diff
@@ -264,3 +264,10 @@ function remake(thing::AbstractEnsembleProblem; kwargs...)
     en_kwargs = [k for k in kwargs if first(k) ∈ fieldnames(T)]
     T(remake(thing.prob; setdiff(kwargs, en_kwargs)...); en_kwargs...)
 end
+
+function remake!(thing::T, args...; kwargs...) where T
+    newthing = remake(thing, args...; kwargs...)
+    for field in fieldnames(T)
+        @set! thing.field = newthing.field
+    end
```
Comment (ChrisRackauckas) [inline src/remake.jl:272]: why is this needed?

**Comment by ChrisRackauckas** [comment]
Needs tests

**Comment by xtalax** [comment]
Yep, was just waiting for comment

**Comment by xtalax** [comment]
might want to revert this -  this had not been debugged as I ran in to trouble with a compilation error in ODE
